prime-router/src/main/kotlin/Report.kt

```diff
@@ -1608,13 +1582,24 @@ class Report : Logging {
             header: WorkflowEngine.Header,
             metadata: Metadata? = null,
         ): String {
-            return if (header.reportFile.bodyUrl != null) {
-                BlobAccess.BlobInfo.getBlobFilename(header.reportFile.bodyUrl)
+            return if (header.receiver?.topic?.isSendOriginal == true) {
```

Comment:
Maybe as a follow up, but we might want to collapse/rename this with other `formExternalFilename` since they have different logic and potentially one makes a DB call